### PR TITLE
Add scaffold framework changes for LINQPad integration

### PR DESCRIPTION
**Pull request by MaceWindu**
will be merged after linqpad refactoring done to catch all required changes

Various changes, required for scaffold framework integration into LINPad driver:

- add internal mechanism to track data model identifiers changes and access them using new interceptor event `AfterSourceCodeGenerated`
- several small visibility changes to various scaffold framework members
- fix location of database name property in function model
- add foreign key name to association model

LINQPad PR: https://github.com/linq2db/linq2db.LINQPad/pull/82

### CODE REVIEW

File: Source/LinqToDB.Tools/CodeModel/Types/WellKnownTypes.cs [hunk 1]
```diff
@@ -705,19 +705,24 @@ public static class Data
...
 				/// <summary>
 				/// DataConnectionExtensions.QueryProcAsync method reference.
 				/// </summary>
-				public static CodeIdentifier DataConnectionExtensions_QueryProcAsync { get; } = new CodeIdentifier(nameof(global::LinqToDB.Data.DataConnectionExtensions.QueryProcAsync), true);
+				public static CodeIdentifier DataConnectionExtensions_QueryProcAsync   { get; } = new CodeIdentifier(nameof(global::LinqToDB.Data.DataConnectionExtensions.QueryProcAsync), true);
+
+				/// <summary>
+				/// <see cref="DataConnection.CommandTimeout"/> property reference.
+				/// </summary>
+				public static CodeReference DataConnection_CommandTimeout { get; } = PropertyOrField((DataConnection dc) => dc.CommandTimeout, false);
```
Comment (viceroypenguin) [inline Source/LinqToDB.Tools/CodeModel/Types/WellKnownTypes.cs:725]: `CodeIdentifier`s above are all aligned on `{ get; }`, while this line and lower are not. Not perferential myself, but others are.

File: Source/LinqToDB.Tools/Scaffold/DataModel/DataModelLoader.Entities.cs [hunk 2]
```diff
@@ -288,15 +288,16 @@ private void BuildEntityColumns(TableLikeObject table, EntityModel entity)
...
 			NormalizationOptions settings,
 			ISet<string>         defaultSchemas)
 		{
 			var name = otherTable.Name;
 
 			// T4 compatibility mode use logic, similar to one, used by old T4 templates
-			if (settings.Transformation == NameTransformation.Association)
+			if (fkName != null && settings.Transformation == NameTransformation.Association)
 			{
+				System.Diagnostics.Debugger.Launch();
```
Comment (viceroypenguin) [inline Source/LinqToDB.Tools/Scaffold/DataModel/DataModelLoader.Entities.cs:300]: Should this be left in?

**Review by sdanyliv — APPROVED** [review]

**Pull request merged**